version: '3.8'

services:
  redis:
    image: redis:7
    container_name: auth_redis
    ports:
      - "6379:6379"
    volumes:
      - ./redis-data:/data
    command: redis-server --appendonly yes
    networks:
      - app_network

  auth_db:
    image: postgres:15
    container_name: auth_db
    environment:
      POSTGRES_USER: auth_user
      POSTGRES_PASSWORD: auth123
      POSTGRES_DB: auth_db
    ports:
      - "5432:5432"
    volumes:
      - ./postgres-data:/var/lib/postgresql/data
    networks:
      - app_network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U auth_user -d auth_db"]
      interval: 10s
      timeout: 5s
      retries: 5

networks:
  app_network:
    driver: bridge